src/main/java/me/svdev/mscategory/service/CategoryService.java

```diff
@@ -0,0 +1,39 @@
+package me.svdev.mscategory.service;
+
+import lombok.RequiredArgsConstructor;
+import me.svdev.mscategory.dao.entity.Category;
+import me.svdev.mscategory.dao.repository.CategoryRepository;
+import me.svdev.mscategory.mapper.CategoryMapper;
+import me.svdev.mscategory.model.response.CategoriesResponse;
+import me.svdev.mscategory.model.dto.CategoryDto;
+import org.springframework.stereotype.Service;
+
+import static me.svdev.mscategory.mapper.CategoryMapper.mapEntityToDto;
+
+@RequiredArgsConstructor
+@Service
+public class CategoryService {
+
+    private final CategoryRepository categoryRepository;
+
+    public CategoriesResponse getCategories() {
+        return new CategoriesResponse(categoryRepository.findAll().stream().map(CategoryMapper::mapEntityToDto).toList());
+    }
+
+    public CategoryDto getCategory(Long id) {
+        return mapEntityToDto(getCategoryIfExists(id));
+    }
+
+//    public void saveCategory(CategoryRequest categoryRequest) {
+//        var category = Category.builder()
+//                .name(categoryRequest.getName())
+//                .isVisible(true)
+//                .isFavorite(categoryRequest.getIsFavorite())
+//                .build();
+//        categoryRepository.save(category);
+//    }
+
+    private Category getCategoryIfExists(Long id) {
+        return categoryRepository.findById(id).orElseThrow(RuntimeException::new);
```

Comment:
bad practice to throw exceptions from Java, create your custom exception, for example NotFoundException and add message to exception. Also you will be need error handling